packages/sdk/src/adapters/multipeer/rules.ts

```diff
@@ -911,11 +912,69 @@ export const Rules: { [id in Payloads.PayloadType]: Rule } = {
     'set-sound-state': {
         ...DefaultRule,
         synchronization: {
-            stage: 'create-animations',
+            stage: 'active-sound-instances',
             before: 'ignore',
             during: 'queue',
             after: 'allow'
+        },
+        session: {
+            ...DefaultRule.session,
+            beforeReceiveFromApp: (
+                session: Session,
+                message: Message<Payloads.SetSoundState>
+            ) => {
+                const syncActor = session.actorSet[message.payload.actorId];
+                if (syncActor) {
+                    syncActor.activeSoundInstances = syncActor.activeSoundInstances || [];
+
+                    const basisTime = Date.now() / 1000.0;
+                    if (message.payload.soundCommand === SoundCommand.Start) {
+                        syncActor.activeSoundInstances.push({ message, basisTime });
+                    } else {
+                        // find the existing message that needs to be updated
+                        const activeSoundInstance = syncActor.activeSoundInstances.filter(
+                            item => item.message.payload.id === message.payload.id).shift();
+
+                        // if sound expired then skip this message completely
+                        if (!activeSoundInstance) {
+                            return undefined;
+                        }
+                        // Remove the existing sound instance (we'll add an updated one below).
+                        syncActor.activeSoundInstances =
+                            (syncActor.activeSoundInstances || []).filter(
```

Comment:
> || [] [](start = 59, length = 6)

minor: `|| []` not needed here, you're initializing syncActor.activeSoundInstances above.